sdk/schema.go

```diff
@@ -55,38 +55,7 @@ type DeploymentResource struct {
 	RequestID       string                 `json:"requestId,omitempty"`
 	ResourceID      string                 `json:"resourceId,omitempty"`
 	ResourceType    string                 `json:"resourceType,omitempty"`
-	ResourcesData   DeploymentResourceData `json:"data,omitempty"`
-}
-
-// DeploymentResourceData - view of the resources/machines in a deployment
-type DeploymentResourceData struct {
-	Memory                      int    `json:"MachineMemory,omitempty"`
-	CPU                         int    `json:"MachineCPU,omitempty"`
-	IPAddress                   string `json:"ip_address,omitempty"`
-	Storage                     int    `json:"MachineStorage,omitempty"`
-	MachineInterfaceType        string `json:"MachineInterfaceType,omitempty"`
-	MachineName                 string `json:"MachineName,omitempty"`
-	MachineGuestOperatingSystem string `json:"MachineGuestOperatingSystem,omitempty"`
-	MachineDestructionDate      string `json:"MachineDestructionDate,omitempty"`
-	MachineGroupName            string `json:"MachineGroupName,omitempty"`
-	MachineBlueprintName        string `json:"MachineBlueprintName,omitempty"`
-	MachineReservationName      string `json:"MachineReservationName,omitempty"`
-	MachineType                 string `json:"MachineType,omitempty"`
-	MachineID                   string `json:"machineId,omitempty"`
-	MachineExpirationDate       string `json:"MachineExpirationDate,omitempty"`
-	Component                   string `json:"Component,omitempty"`
-	Expire                      bool   `json:"Expire,omitempty"`
-	Reconfigure                 bool   `json:"Reconfigure,omitempty"`
-	Reset                       bool   `json:"Reset,omitempty"`
-	Reboot                      bool   `json:"Reboot,omitempty"`
-	PowerOff                    bool   `json:"PowerOff,omitempty"`
-	Destroy                     bool   `json:"Destroy,omitempty"`
-	Shutdown                    bool   `json:"Shutdown,omitempty"`
-	Suspend                     bool   `json:"Suspend,omitempty"`
-	Reprovision                 bool   `json:"Reprovision,omitempty"`
-	ChangeLease                 bool   `json:"ChangeLease,omitempty"`
-	ChangeOwner                 bool   `json:"ChangeOwner,omitempty"`
-	CreateSnapshot              bool   `json:"CreateSnapshot,omitempty"`
+	ResourcesData   map[string]interface{} `json:"data,omitempty"`
```

Comment:
I'm generally not in-favour of loosening the type constraints, especially when parsing JSON. But enforcing the types means that the clients lose the generic APIs for custom blueprint properties and XaaS.